tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/ConstSimplifierBase.scala

```diff
@@ -113,130 +128,93 @@ abstract class ConstSimplifierBase {
-        ValEx(TlaBool(true))(boolTag) // an empty conjunction is true
-      } else if (simpArgs.contains(ValEx(TlaBool(false))(boolTag))) {
-        ValEx(TlaBool(false))(boolTag) // one false make conjunction false
-      } else {
-        OperEx(TlaBoolOper.and, simpArgs: _*)(boolTag)
+      simpArgs match {
+        case Seq()      => ValEx(TlaBool(true))(boolTag) // an empty conjunction is true
+        case Seq(first) => first
+        // one false make conjunction false
+        case _ if simpArgs.contains(ValEx(TlaBool(false))(boolTag)) => ValEx(TlaBool(false))(boolTag)
+        // TRUE /\ x /\ y = x /\ y
+        case _ => OperEx(TlaBoolOper.and, simpArgs: _*)(boolTag)
       }
 
     case OperEx(TlaBoolOper.or, args @ _*) =>
       val simpArgs = args.filterNot {
         _ == ValEx(TlaBool(false))(boolTag)
       }
-      if (simpArgs.isEmpty) {
-        ValEx(TlaBool(false))(boolTag) // an empty disjunction is true
-      } else if (simpArgs.contains(ValEx(TlaBool(true))(boolTag))) {
-        ValEx(TlaBool(true))(boolTag) // one true make disjunction true
-      } else {
-        OperEx(TlaBoolOper.or, simpArgs: _*)(boolTag)
+      simpArgs match {
+        case Seq()      => ValEx(TlaBool(false))(boolTag) // an empty disjunction is false
+        case Seq(first) => first
+        // one true make disjunction true
+        case _ if simpArgs.contains(ValEx(TlaBool(true))(boolTag)) => ValEx(TlaBool(true))(boolTag)
+        // FALSE \/ x \/ y = x \/ y
+        case _ => OperEx(TlaBoolOper.or, simpArgs: _*)(boolTag)
       }
 
-    case OperEx(TlaBoolOper.not, ValEx(TlaBool(b))) =>
-      ValEx(TlaBool(!b))(boolTag)
-
-    case OperEx(TlaBoolOper.not, OperEx(TlaBoolOper.not, underDoubleNegation)) =>
-      underDoubleNegation
-
-    case OperEx(TlaBoolOper.not, OperEx(TlaOper.ne, lhs, rhs)) =>
-      OperEx(TlaOper.eq, lhs, rhs)(boolTag)
-
+    // Evaluate implication of constants
     case OperEx(TlaBoolOper.implies, ValEx(TlaBool(left)), ValEx(TlaBool(right))) =>
       ValEx(TlaBool(!left || right))(boolTag)
+    case OperEx(TlaBoolOper.implies, ValEx(TlaBool(false)), _) => ValEx(TlaBool(true))(boolTag)
+    case OperEx(TlaBoolOper.implies, _, ValEx(TlaBool(true)))  => ValEx(TlaBool(true))(boolTag)
 
-    case OperEx(TlaBoolOper.implies, ValEx(TlaBool(false)), _) =>
-      ValEx(TlaBool(true))(boolTag)
-
-    case OperEx(TlaBoolOper.implies, ValEx(TlaBool(true)), right) =>
-      simplifyShallow(OperEx(TlaBoolOper.not, right)(boolTag))
-
-    case OperEx(TlaBoolOper.implies, _, ValEx(TlaBool(true))) =>
-      ValEx(TlaBool(true))(boolTag)
-
+    // TRUE -> x = x
+    case OperEx(TlaBoolOper.implies, ValEx(TlaBool(true)), right) => right
+    // x -> FALSE = !x
     case OperEx(TlaBoolOper.implies, lhs, ValEx(TlaBool(false))) =>
       simplifyShallow(OperEx(TlaBoolOper.not, lhs)(boolTag))
 
+    // Evaluate equivalence between constants
     case OperEx(TlaBoolOper.equiv, ValEx(TlaBool(left)), ValEx(TlaBool(right))) =>
```

Comment:
I was under the impression that one of the passes transforms `<=>` into `=`. Do any of these ever fire?

Reply:
I checked and they don't, at least not from our integration test suite. I got rid of them, thanks!